cmd/blocking-issue-creator/main.go

```diff
@@ -98,71 +106,76 @@ func main() {
 			return nil
 		}
 
-		var branchTokens []string
-		body := fmt.Sprintf("The following branches are being fast-forwarded from the current development branch (%s) as placeholders for future releases. No merging is allowed into these release branches until they are unfrozen for production release.\n\n", repoInfo.Branch)
-		for _, branch := range branches.List() {
-			body += fmt.Sprintf(" - `%s`\n", branch)
-			branchTokens = append(branchTokens, fmt.Sprintf("branch:%s", branch))
-		}
-		body += "\nContact the [Test Platform](https://coreos.slack.com/messages/CBN38N3MW) or [Automated Release](https://coreos.slack.com/messages/CB95J6R4N) teams for more information."
-		title := fmt.Sprintf("Future Release Branches Frozen For Merging | %s", strings.Join(branchTokens, " "))
-
-		query := fmt.Sprintf("is:issue state:open label:\"tide/merge-blocker\" repo:%s/%s author:%s", repoInfo.Org, repoInfo.Repo, o.githubLogin)
-		sort := "updated"
-		// We will make sure that the first issue in the list will be with the most recent update.
-		ascending := false
-		issues, err := client.FindIssues(query, sort, ascending)
-		if err != nil {
-			logger.WithError(err).Error("Failed to search for open issues.")
+		if err := operateOnIssues(client, o.githubLogin, repoInfo, branches, logger); err != nil {
 			failed = true
+			return nil
 		}
 
-		if len(issues) > 1 {
-			logger.Warnf("Found more than one merge blocking issue by the bot: %v", len(issues))
-			for _, issue := range issues[1:] {
-				if err := client.CloseIssue(repoInfo.Org, repoInfo.Repo, issue.Number); err != nil {
-					logger.WithError(err).Error("Failed to close issue.")
-					failed = true
-					return nil
-				}
-				logger.WithField("number", issue.Number).Info("Closed extra issue.")
-			}
-		}
+		return nil
+	}); err != nil || failed {
+		logrus.WithError(err).Fatal("Could not publish merge blocking issues.")
+	}
+}
 
-		// we have an existing issue that needs to be up to date
-		if len(issues) != 0 {
-			logger = logger.WithField("merge-blocker", issues[0])
-			existing := issues[0]
+func operateOnIssues(client githubClient, githubLogin string, repoInfo *config.Info, branches sets.String, logger *logrus.Entry) error {
```

Comment:
"operate on issues" is not a useful name. I would expect such name for a function that iterates over all issues and does a custom "some operation" on each, where the custom operation is e.g. a parameter function or something similar.